Cargo.toml

```diff
@@ -74,6 +74,7 @@ pin-project-lite = "0.2.9"
 url = "2.4.0"
 once_cell = "1.18.0"
 chrono = "0.4.31"
+trivial_generic = { path = "./trivial_generic" }
```

Comment:
can it live inside micro-rdk? maybe a separate macro folder?

Reply:
it can't, I think? proc-macros need to be their own crate

Reply:
I interpreted this request as moving into src, which I was able to do, but I'm concerned it's no longer usable by modular drivers?